src/core/raster/qgssinglebandpseudocolorrenderer.cpp

```diff
@@ -106,6 +297,8 @@ QgsRasterRenderer* QgsSingleBandPseudoColorRenderer::create( const QDomElement&
   // TODO: add _readXML in superclass?
   r->setClassificationMin( elem.attribute( QStringLiteral( "classificationMin" ), QStringLiteral( "NaN" ) ).toDouble() );
   r->setClassificationMax( elem.attribute( QStringLiteral( "classificationMax" ), QStringLiteral( "NaN" ) ).toDouble() );
+  r->setClassificationMode( static_cast< Mode >( elem.attribute( QStringLiteral( "classificationMode" ), QStringLiteral( "NaN" ) ).toInt() ) );
+  r->setClassificationNbClasses( elem.attribute( QStringLiteral( "classificationNbClasses" ), QStringLiteral( "NaN" ) ).toInt() );
```

Comment:
NaN only makes sense with floats/doubles, but not for ints / enums. Why not using the default values used in constructor (continuous / 5 classes) ?